packages/engine/src/HedronEngine/importSketchModule.ts

```diff
@@ -1,7 +1,69 @@
 import { Result } from './types'
-import { NodeTypes, SketchConfig, SketchModule, SketchModuleItem } from '@store/types'
+import {
+  NodeTypes,
+  SketchConfigRaw,
+  SketchConfigImported,
+  SketchModule,
+  SketchModuleItem,
+  SketchConfigParamImported,
+  SketchConfigParam,
+} from '@store/types'
 import { createUniqueId } from '@utils/createUniqueId'
 
+const ensureParamImported = (
+  param: SketchConfigParam,
+  groupIndex: number | null,
+): SketchConfigParamImported => {
+  const valueType =
+    'valueType' in param && param.valueType !== undefined ? param.valueType : NodeTypes.Number
+
+  return {
+    ...param,
+    title: param.title ?? param.key,
+    groupIndex,
+    valueType,
+  } as SketchConfigParamImported
+}
+
+interface ProcessConfigOptions {
+  fallBackTitle: string
+}
+
+const processConfig = (
+  config: SketchConfigRaw,
+  { fallBackTitle }: ProcessConfigOptions,
+): SketchConfigImported => {
+  const groupInfo: SketchConfigImported['groupInfo'] = []
+
+  let groupIndex = -1
+  const flattenedParams: SketchConfigParamImported[] = []
+
+  if (config.params) {
+    config.params.forEach((param) => {
+      if ('params' in param) {
+        groupIndex++
+
+        groupInfo[groupIndex] = {
+          groupTitle: param.groupTitle ?? `Group ${groupIndex}`,
+        }
+
+        flattenedParams.push(...param.params.map((p) => ensureParamImported(p, groupIndex)))
+      } else {
+        flattenedParams.push(ensureParamImported(param, null))
+      }
+    })
+  }
+
+  const processedConfig: SketchConfigImported = {
+    ...config,
+    title: config.title ?? fallBackTitle,
+    params: flattenedParams,
+    groupInfo,
+  }
+
+  return processedConfig
```

Comment:
this is adding some type safety to how we import stuff. We have an expected raw config format and an expected imported format.